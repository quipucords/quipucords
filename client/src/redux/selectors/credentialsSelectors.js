import { createSelector } from 'reselect';
import apiTypes from '../../constants/apiConstants';

/**
 * Map a credential array to a consumable dropdown format
 */
const credentials = state => state.credentials.view.credentials;

const credentialsDropdownSelector = createSelector(
  [credentials],
  creds =>
    (creds || []).map(cred => ({
      title: cred[apiTypes.API_RESPONSE_CREDENTIAL_NAME],
      value: cred[apiTypes.API_RESPONSE_CREDENTIAL_ID],
      type: cred[apiTypes.API_RESPONSE_CREDENTIAL_CRED_TYPE]
    }))
);

const makeCredentialsDropdownSelector = () => credentialsDropdownSelector;

const credentialsSelectors = {
  credentialsDropdown: credentialsDropdownSelector,
  makeCredentialsDropdown: makeCredentialsDropdownSelector
};

export {
  credentialsSelectors as default,
  credentialsSelectors,
  credentialsDropdownSelector,
  makeCredentialsDropdownSelector
};
