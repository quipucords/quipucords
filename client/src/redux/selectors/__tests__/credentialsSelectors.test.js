import credentialsSelectors from '../credentialsSelectors';
import apiTypes from '../../../constants/apiConstants';

describe('CredentialsSelectors', () => {
  it('Should return specific selectors', () => {
    expect(credentialsSelectors).toMatchSnapshot('selectors');
  });

  it('Should map a response to a consumable dropdown format', () => {
    const state = {
      credentials: {
        view: {
          credentials: [
            {
              [apiTypes.API_RESPONSE_CREDENTIAL_NAME]: 'Lorem',
              [apiTypes.API_RESPONSE_CREDENTIAL_CRED_TYPE]: 'network',
              [apiTypes.API_RESPONSE_CREDENTIAL_ID]: 54
            },
            {
              [apiTypes.API_RESPONSE_CREDENTIAL_NAME]: 'Ipsum',
              [apiTypes.API_RESPONSE_CREDENTIAL_CRED_TYPE]: 'vcenter',
              [apiTypes.API_RESPONSE_CREDENTIAL_ID]: 1
            },
            {
              [apiTypes.API_RESPONSE_CREDENTIAL_NAME]: 'Dolor',
              [apiTypes.API_RESPONSE_CREDENTIAL_CRED_TYPE]: 'satellite',
              [apiTypes.API_RESPONSE_CREDENTIAL_ID]: 200
            }
          ]
        }
      }
    };

    expect(credentialsSelectors.credentialsDropdown(state)).toMatchSnapshot('credentialsDropdown');
  });
});
