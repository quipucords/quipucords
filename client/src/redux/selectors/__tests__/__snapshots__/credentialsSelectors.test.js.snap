// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`credentialsDropdown 1`] = `
Array [
  Object {
    "title": "Lorem",
    "type": "network",
    "value": 54,
  },
  Object {
    "title": "Ipsum",
    "type": "vcenter",
    "value": 1,
  },
  Object {
    "title": "Dolor",
    "type": "satellite",
    "value": 200,
  },
]
`;

exports[`selectors 1`] = `
Object {
  "credentialsDropdown": [Function],
  "makeCredentialsDropdown": [Function],
}
`;
